pkg/kubernetes/kubernetes.go

```diff
@@ -64,7 +64,7 @@ func (k *Kubernetes) Reconcile(raw map[string]interface{}, objectspecs ...string
 	}
 	for _, d := range docs {
 		m := objx.New(d)
-		if k != nil {
+		if k != nil && !m.Has("metadata.namespace") {
 			m.Set("metadata.namespace", k.Spec.Namespace)
```

Comment:
Is their a usecase where you wouldn't want the namespace explicitly set in the `spec.json`? Should it instead just be an error if someone doesn't set this value?

Reply:
Yes there is. Some operators require certain objects to be put into a namespace they watch, even if they are related to a single application.

Furthermore it would be too opinionated IMHO to allow only single-namespace apps